src/helm/proxy/clients/huggingface_client.py

```diff
@@ -93,8 +97,10 @@ def serve_request(self, raw_request: Dict[str, Any]):
         if not raw_request["echo_prompt"]:
             sequences = [sequence[len(encoded_input.input_ids[0]) :] for sequence in sequences]
 
-        # TODO: Get rid of the extra tokenization?
-        all_tokens = [self.tokenizer.convert_ids_to_tokens(sequence) for sequence in sequences]
+        all_tokens = [
+            [self.tokenizer.decode(token) for token in sequence_tokens] 
+            for sequence_tokens in sequences
+        ]
         all_decoded_text = self.tokenizer.batch_decode(sequences)
```

Comment:
I am a bit confused here, why do we decode twice? Also, I saw that you removed `convert_ids_to_tokens`, would `convert_tokens_to_string(convert_ids_to_tokens(token))` be what you are looking for?

Reply:
And yes I also originally had `convert_tokens_to_string(convert_ids_to_tokens(token))`, but I think that does the same thing as `decode`!